Atomic/util.py

```diff
@@ -802,102 +801,79 @@ def load_scan_result_file(file_name):
         while time_out < 10.5: # Ten second attempt to get a lock
             try:
-                fcntl.flock(install_data_file, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                fcntl.flock(f, fcntl.LOCK_EX | fcntl.LOCK_NB)
                 f_lock = True
                 break
             except IOError:
                 time.sleep(.5)
                 time_out += .5
         if not f_lock:
-            raise ValueError("Unable to get file lock for {}".format(ATOMIC_INSTALL_JSON))
-
-    def release_lock():
-        fcntl.flock(install_data_file, fcntl.LOCK_UN)
-
-    def wrapper(*args, **kwargs):
-        get_lock()
-        ret = func(*args, **kwargs)
-        release_lock()
-        return ret
-
-    return wrapper
+            raise ValueError("Unable to get file lock for {}".format(lock_file_name))
 
+        # Call the user code
+        yield
+        # Now unlock
+        fcntl.flock(f, fcntl.LOCK_UN)
 
+# Records additional data for containers outside of the native storage (docker/ostree)
 class InstallData(object):
-    if not os.path.exists(ATOMIC_INSTALL_JSON):
-        open(ATOMIC_INSTALL_JSON, 'a').close()
-
-    install_file_handle = open(ATOMIC_INSTALL_JSON, 'r')
-
-    @staticmethod
-    def _read_install_data(file_handle):
-        try:
-            return json.loads(file_handle.read())
-        except ValueError:
-            return {}
-
-    @classmethod
-    def _write_install_data(cls, new_data):
-        install_data = cls._read_install_data(cls.install_file_handle)
-        for x in new_data:
-            install_data[x] = new_data[x]
-        temp_file = tempfile.NamedTemporaryFile(mode='w', delete=False)
-        json.dump(install_data, temp_file)
-        temp_file.close()
-        shutil.move(temp_file.name, ATOMIC_INSTALL_JSON)
 
     @classmethod
-    @file_lock
     def read_install_data(cls):
-        if os.path.exists(ATOMIC_INSTALL_JSON):
-            read_data = cls._read_install_data(cls.install_file_handle)
-            return read_data
-        return {}
+        with file_lock(ATOMIC_INSTALL_JSON):
+            try:
+                with open(ATOMIC_INSTALL_JSON, 'r') as f:
+                    # Backwards compatibilty - we previously created an empty file explicitly;
+                    # see https://github.com/projectatomic/atomic/pull/966
+                    if os.fstat(f.fileno()).st_size == 0:
+                        return {}
+                    return json.load(f)
+            except IOError as e:
+                if e.errno == errno.ENOENT:
+                    return {}
+                raise e
 
     @classmethod
-    @file_lock
     def write_install_data(cls, new_data):
-        cls._write_install_data(new_data)
+        install_data = cls.read_install_data()
+        with file_lock(ATOMIC_INSTALL_JSON):
+            for x in new_data:
+                install_data[x] = new_data[x]
+                temp_file = tempfile.NamedTemporaryFile(mode='w', delete=False)
```

Comment:
Seems like this line and the rest below are supposed to be unindented by one level?